frame/contracts/src/storage.rs

```diff
@@ -205,14 +205,14 @@ impl<T: Config> ContractInfo<T> {
 	///
 	/// You must make sure that the contract is also removed when queuing the trie for deletion.
 	pub fn queue_trie_for_deletion(&self) -> DispatchResult {
-		<DeletionQueue<T>>::try_append(DeletedContract { trie_id: self.trie_id.clone() })
-			.map_err(|_| <Error<T>>::DeletionQueueFull.into())
+		DeletionQueue::<T>::load().insert(DeletedContract { trie_id: self.trie_id.clone() });
+		Ok(())
 	}
 
 	/// Calculates the weight that is necessary to remove one key from the trie and how many
 	/// of those keys can be deleted from the deletion queue given the supplied queue length
 	/// and weight limit.
-	pub fn deletion_budget(queue_len: usize, weight_limit: Weight) -> (Weight, u32) {
```

Comment:
This budget seems wrong now
```rust
		let decoding_weight = weight_per_queue_item.saturating_mul(queue_len as u64);
```
We should not depend on the length of the queue to calculate the budget since it can now be unbounded
Will revisit this logic